### DISCUSSION TITLE
What about reducing the number of messages by filtering based on the popularity?

### DISCUSSION THREAD

**Discussion by tahaygun** [jiggy-ai/hn_summary#10] [Ideas]
Hi, just jumped in the channel. Everything looks pretty good so far but I find the number of messages coming in incredibly too much. It is impossible to follow what's going on. So maybe only the posts above X up votes should be sent to the telegram channel? WDYT?

**Comment by wskish** [comment]
Yes currently it is surfacing everything that appears on the /topstories endpoint which seems to be 500.

One simple strategy is to limit to top-N stories, like 250.  This should be effective since the probability of a story making it into (e.g.) top 250 should be much lower than it making into top 500.    There are probably some other simple adaptive strategies that would work well.

We are also thinking about a version of this where users can personalize their interaction with the bot, potentially setting individual thresholds or even topic filters using semantic search.  

As a first step I will play with using a higher top-N threshold to see how that feels.

**Comment by wskish** [comment]
Updated to only look at top 400 stories instead of previous top 500.  Will make further adjustments once I get a feel for the impact and think about alternatives a bit more.

**Comment by tahaygun** [comment]
Thanks for the update and quick response. Personalization would be the best option for sure!
